### ISSUE TITLE
std.parseYaml wraps result in an array when string values contain ---

### ISSUE THREAD

**Issue by SamuraiPrinciple** [google/jsonnet#1147]
Example:

```
[
  std.parseYaml('a: good--'),
  std.parseYaml('a: bad---'),
]
```

Produces:

```
[
   {
      "a": "good--"
   },
   [
      {
         "a": "bad---"
      }
   ]
]
```

**Comment by johnbartholomew** [comment]
Hi. Thanks for the bug report. Could you confirm exactly which version of jsonnet you see this behaviour with? If you've built it yourself, specifying which git commit (also how you build it would be good, since it can be configured with a bundled YAML parsing library or configured to use one from your system). If you're using a binary from somewhere else, running `jsonnet --version` should give some version information.

So far I have not been able to reproduce this myself, so I am probably running a different version of jsonnet.

**Comment by SamuraiPrinciple** [comment]
Hi. Many thanks for the quick reply.

Initially, I've observed the issue on argocd which seems to be using go-jsonnet.

If I highjack one of the examples on jsonnet.org website, I'm able to reproduce it as well:

<img width="2672" alt="Screenshot 2024-04-01 at 15 18 05" src="https://github.com/google/jsonnet/assets/621249/25c0ad42-5063-4c7b-bad4-eb6c6c36914e">

I'm also able to reproduce it with vscode jsonnet language server plugin (that seems to be using go-jsonnet too) https://marketplace.visualstudio.com/items?itemName=Grafana.vscode-jsonnet:

<img width="2672" alt="Screenshot 2024-04-01 at 15 24 07" src="https://github.com/google/jsonnet/assets/621249/baa683f2-b27e-4900-9f8f-0e38fb6a9bc2">

I have just realised that my local jsonnet version was actually go-jsonnet (and it's also affected) which leads me to the conclusion that I've got it wrong way around - it's most likely that only the go version is affected.
